app/models/backend_api.rb

```diff
@@ -107,11 +108,6 @@ def schedule_deletion
     DeleteObjectHierarchyWorker.perform_later(self)
   end
 
-  def set_private_endpoint
-    return if account.provider_can_use?(:api_as_product)
-    self.private_endpoint ||= default_api_backend
-  end
-
   def set_port_private_endpoint
     Proxy::PortGenerator.new(self).call(:private_endpoint)
   end
```

Comment:
We do not need this anymore either because now the field 'private_endpoint' is empty ('blank') and the 1st thing that this line does is just returning in this scenario 😄 

https://github.com/3scale/porta/blob/99901577f26e32c57a086f7c3c2b0e0d9240ca96/app/models/proxy.rb#L689-L696

Reply:
Does it depends directly on this PR or can it be extracted into a parallel one?

Reply:
many things of this PR can be extracted to other PRs 😄 like this 👍🏼 😉

Reply:
I don't see why we'd remove the `set_port_private_endpoint` callback. I agree we no longer need the `set_private_endpoint` one, but, still, this does NOT mean `BackendApi`'s `private_endpoint` will be empty/`nil`. It can still be set manually, like in https://github.com/3scale/porta/blob/2d3d1f33ffe144b8e4f0006fdbf22e988341fae3/app/models/account/provider_methods.rb#L346

Perhaps there was a confusion between `BackendApi`'s `private_endpoint` and `Proxy`'s `private_endpoint`?